packages/static_shock_cli/bin/static_shock_cli.dart

```diff
@@ -122,16 +115,83 @@ class ServeCommand extends Command {
   }
 }
 
-// class ValidateCommand extends Command {
-//   @override
-//   final name = "validate";
-//
-//   @override
-//   final description = "Inspects the current working directory and validates its structure as a Static Shock project.";
-//
-//   @override
-//   Future<void> run() async {
-//     // TODO:
-//     print("Validating Static Shock project!");
-//   }
-// }
+class UpgradeCommand extends Command {
+  @override
+  final name = "upgrade";
+
+  @override
+  final description = "Upgrades your version of static_shock_cli to the latest from Pub.";
+
+  @override
+  Future<void> run() async {
+    final pubUpdater = PubUpdater();
+
+    _log.info("Checking for newer versions of static_shock_cli...");
```

Comment:
Is it a good idea for us to run a possibly long-running network call without any indication to the user that something is happening?

Reply:
Yes we should -- I don't want that kinda stuff in my nice, tidy CLI output on every single command. Silently checking is industry standard practice.

What we should do is run the network call in the background in parallel with the requested command so it doesn't take extra time. Bonus points for killing the network check if the requested command finishes first.

Reply:
Ok, we can try to make that happen. 

For clarity, you're saying that the Flutter tools version check isn't industry standard? I believe Flutter checks the version every time, or at least at some rate, and when it does that, it doesn't run your desired command in parallel, it makes you wait for the version check.

Reply:
I said that silently checking is the industry standard for CLI tools, not running the check in parallel (that would be for extra credit). I don't believe I've ever since the Flutter CLI report that it's checking for a new version; just that it's found one (when it does).